### PR TITLE
Adding apache camel k blog post

### PR DESCRIPTION
**Pull request by matzew**
## Proposed Changes 

- Blog of Apache Camel K and Knative Eventing

### CODE REVIEW

**Comment by matzew** [comment]
close, reopen, since CLA check was _wrongly_ failing

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+## Why Apache Camel K?
+
+The [Apache Camel](https://camel.apache.org/){:target="_blank"} is a popular Open Source integration framework that empowers you to quickly and easily integrate various systems consuming or producing data. With [Apache Camel K](https://camel.apache.org/camel-k/latest){:target="_blank"} the project provides a lightweight integration framework built from Apache Camel that runs natively on Kubernetes and is specifically designed for serverless and microservice architectures.
+
+The Camel K framework does also support Knative, allowing developers to [bind](https://camel.apache.org/camel-k/latest/kamelets/kamelets-user.html#kamelets-usage-binding){:target="_blank"} any Kamelet to a Knative component. A Kamelet can act as "source" of data or alternatively as "sink". There are several Kamelets available for integrating and connecting to 3rd party services or products, such as Amazon Web Services (AWS), Google Cloud or even tradition message systems like AMQP 1.0 or JMS brokers like Apache Artemis. The full list of Kamelets can be found [here](https://camel.apache.org/camel-kamelets/latest/index.html){:target="_blank"}.
```
Comment (dsimansk) [inline blog/docs/articles/knative-meets-apache-camel.md:9]:
```suggestion
The Camel K framework does also support Knative, allowing developers to [bind](https://camel.apache.org/camel-k/latest/kamelets/kamelets-user.html#kamelets-usage-binding){:target="_blank"} any Kamelet to a Knative component. A Kamelet can act as "source" of data or alternatively as "sink". There are several Kamelets available for integrating and connecting to 3rd party services or products, such as Amazon Web Services (AWS), Google Cloud or even tradition message systems like AMQP 1.0 or JMS brokers like Apache Artemis. The full list of Kamelets can be found in the [documentation](https://camel.apache.org/camel-kamelets/latest/index.html){:target="_blank"}.
```
@matzew wdyt about this suggestion?

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+## Installation
+
+The [Installation](https://camel.apache.org/camel-k/next/installation/installation.html) from Apache Camel K offers a few choices, such as CLI, Kustomize, OLM or Helm, like:
+
```
Comment (dsimansk) [inline blog/docs/articles/knative-meets-apache-camel.md:14]: I'd end previous sentece with `.` dot and add sub-title `Example of Helm installation:`

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  name: demo-broker
+```
+
+Now event producers can send events to it and event consumers can receive events.
+
+## Using Kamelets as Event Sources
+
+In order to bind a Kamelet to a Knative component, like the above broker, we are using the `Pipe` API. A Pipe allows to declaratively move data from a system described by a Kamelet _towards_ a Knative destination **or** _from_ a Knative destination to another (external) system described by a Kamelet.
+
+Below is a `Pipe` that uses a ready-to-use `Kamelet`, the `timer-source`
```
Comment (dsimansk) [inline blog/docs/articles/knative-meets-apache-camel.md:40]:
```suggestion
Below is a `Pipe` that uses a ready-to-use `Kamelet`, a `timer-source`
```
?

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 1 above)
Comment (creydr) [inline blog/docs/articles/knative-meets-apache-camel.md:9]:
```suggestion
The Camel K framework also supports Knative, allowing developers to [bind](https://camel.apache.org/camel-k/latest/kamelets/kamelets-user.html#kamelets-usage-binding){:target="_blank"} any Kamelet to a Knative component. A Kamelet can act as "source" of data or alternatively as "sink". There are several Kamelets available for integrating and connecting to 3rd party services or products, such as Amazon Web Services (AWS), Google Cloud or even tradition message systems like AMQP 1.0 or JMS brokers like Apache Artemis. The full list of Kamelets can be found [here](https://camel.apache.org/camel-kamelets/latest/index.html){:target="_blank"}.
```

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      name: log-sink
+```
+
+The `demo-broker` is referenced as the `source` of the `Pipe` and within the `properties` we define which CloudEvent `type` we are interested in. On a matching CloudEvent, the event is routed to the referenced `sink`. In this example we are using a simple `log-sink` Kamelet, which will just print the received data on its standard out log.
+
+> NOTE: In order for the above to work, the Apache Camel K operator will indeed create a Knative `Trigger` from the `Pipe` data, where the `spec.broker` will match our `demo-broker` and the `spec.filter.attributes.types` field will be set to `com.corp.my.timer.source` to ensure only matching CloudEvent types are being forwarded.
+
+## Conclusion
+
+With Apache Camel K the Knative Eventing ecosystem benefits from a huge number of predefined `Kamelet`s for integration with a lot of services and products. Sending events from Google Cloud to AWS is possible. Knative Eventing acts as the heart of the routing, with the Knative `Broker` and `Trigger` APIs as the Event Mesh for your Kubernetes cluster!
```
Comment (dsimansk) [inline blog/docs/articles/knative-meets-apache-camel.md:96]: I'd consider formatting Knative resources without inline code backticks. It feels a bit more natural as a conclusion text, when I check the rendered preview.

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+## Installation
+
+The [Installation](https://camel.apache.org/camel-k/next/installation/installation.html) from Apache Camel K offers a few choices, such as CLI, Kustomize, OLM or Helm, like:
+
+```
+$ helm repo add camel-k https://apache.github.io/camel-k/charts/
+$ helm install my-camel-k camel-k/camel-k
+```
+
+Beside Camel K we also need to have Knative Eventing installed, as discussed [here](https://knative.dev/docs/install/yaml-install/eventing/install-eventing-with-yaml/){:target="_blank"}.
```
Comment (creydr) [inline blog/docs/articles/knative-meets-apache-camel.md:20]:
```suggestion
Besides Camel K we also need to have Knative Eventing installed, as described in the [documentation](https://knative.dev/docs/install/yaml-install/eventing/install-eventing-with-yaml/){:target="_blank"}.
```

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+In order to bind a Kamelet to a Knative component, like the above broker, we are using the `Pipe` API. A Pipe allows to declaratively move data from a system described by a Kamelet _towards_ a Knative destination **or** _from_ a Knative destination to another (external) system described by a Kamelet.
+
+Below is a `Pipe` that uses a ready-to-use `Kamelet`, the `timer-source`
+
+```yaml
+apiVersion: camel.apache.org/v1
+kind: Pipe
+metadata:
+  name: beer-source-pipe
```
Comment (creydr) [inline blog/docs/articles/knative-meets-apache-camel.md:46]:
```suggestion
  name: timer-source-pipe
```

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      kind: Broker
+      apiVersion: eventing.knative.dev/v1
+      name: demo-broker
+```
+
+The `timer-source` Kamelet is referenced as the `source` of the `Pipe` and sends periodically (default is `1000ms`) the value of its `message` property to the outbound `sink`. Here we use the Knative Broker, which accepts CloudEvents. The conversion of the message payload to CloudEvents format is done by Apache Camel for us. On the `sink` we can also define the `type` of the CloudEvent to be send.
+
+## Using Kamelets as Event Consumers
+
+In order to consume message from the Knative broker, using Apache Camel K we need a different `Pipe` where the above broker acts as the source of events and a Kamelet is used as sink to receive the CloudEvents:
```
Comment (creydr) [inline blog/docs/articles/knative-meets-apache-camel.md:68]:
```suggestion
In order to consume messages from the Knative broker, using Apache Camel K, we need a different `Pipe` where the above Broker acts as the source of events and a Kamelet is used as sink to receive the CloudEvents: 
```

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+  sink:
+    ref:
+      kind: Kamelet
+      apiVersion: camel.apache.org/v1
+      name: log-sink
+```
+
+The `demo-broker` is referenced as the `source` of the `Pipe` and within the `properties` we define which CloudEvent `type` we are interested in. On a matching CloudEvent, the event is routed to the referenced `sink`. In this example we are using a simple `log-sink` Kamelet, which will just print the received data on its standard out log.
+
+> NOTE: In order for the above to work, the Apache Camel K operator will indeed create a Knative `Trigger` from the `Pipe` data, where the `spec.broker` will match our `demo-broker` and the `spec.filter.attributes.types` field will be set to `com.corp.my.timer.source` to ensure only matching CloudEvent types are being forwarded.
```
Comment (dsimansk) [inline blog/docs/articles/knative-meets-apache-camel.md:92]: The note isn't rendered as a highlight, i.e. gray-ish color range right now. But there's `!!! note` notation that pops more. 

An example from other blog post: https://deploy-preview-5992--knative.netlify.app/blog/articles/demystifying-activator-on-path/

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+kind: Pipe
+metadata:
+  name: beer-source-pipe
+spec:
+  source:
+    ref:
+      kind: Kamelet
+      apiVersion: camel.apache.org/v1
+      name: timer-source
+    properties:
```
Comment (creydr) [inline blog/docs/articles/knative-meets-apache-camel.md:53]: Maybe you add a quick note, that the `properties` field is specific for the referenced kind (from .spec.source.ref). As you later use the same for the Trigger filter (line 81 & 82)

**Review by creydr — CHANGES_REQUESTED** [review]
Great to have a blog about this! :muscle: 

Left some comments. Maybe you can also change from "we" to "you" (https://github.com/knative/docs/blob/main/contribute-to-docs/style-guide/voice-and-language.md#address-the-reader-as-you-not-we) :shrug:

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 8 above)
Comment (matzew) [reply blog/docs/articles/knative-meets-apache-camel.md:92]: Ah, thanks!

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 9 above)
Comment (matzew) [reply blog/docs/articles/knative-meets-apache-camel.md:53]: I have a sentence below that example.

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+metadata:
+  name: log-sink-pipe
+spec:
+  source:
+    ref:
+      kind: Broker
+      apiVersion: eventing.knative.dev/v1
+      name: demo-broker
+    properties:
+      type: com.corp.my.timer.source
```
Comment (pierDipi) [inline blog/docs/articles/knative-meets-apache-camel.md:82]: Is this necessary for a "sink" ?

File: blog/docs/articles/knative-meets-apache-camel.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+The `demo-broker` is referenced as the `source` of the `Pipe` and within the `properties` we define which CloudEvent `type` we are interested in. On a matching CloudEvent, the event is routed to the referenced `sink`. In this example we are using a simple `log-sink` Kamelet, which will just print the received data on its standard out log.
+
+
+!!! note
+
+    In order for the above to work, the Apache Camel K operator will indeed create a Knative `Trigger` from the `Pipe` data, where the `spec.broker` will match our `demo-broker` and the `spec.filter.attributes.types` field will be set to `com.corp.my.timer.source` to ensure only matching CloudEvent types are being forwarded.
+
+## Conclusion
+
+With Apache Camel K the Knative Eventing ecosystem benefits from a huge number of predefined `Kamelet`s for integration with a lot of services and products. Sending events from Google Cloud to AWS is possible. Knative Eventing acts as the heart of the routing, with the Knative Broker and Trigger APIs as the Event Mesh for your Kubernetes cluster!
```
[the remark this answers is not in the document]
Comment (dsimansk) [reply blog/docs/articles/knative-meets-apache-camel.md]: @matzew it's probably fine either way. I'll leave that to your eye to merger or not. :)

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 10 above)
Comment (matzew) [reply blog/docs/articles/knative-meets-apache-camel.md:82]: no, it is optional, w/o that it has a different CloudEvent type (`org.apache....`)

Comment (pierDipi) [reply blog/docs/articles/knative-meets-apache-camel.md:82]: this is sink though? isn't it using the given type already ?

Comment (matzew) [reply blog/docs/articles/knative-meets-apache-camel.md:82]: ah, sorry. I did not see _which_ you commented on. This is (currently) required. I have filed a bug w/ them, last week

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 9 above)
Comment (creydr) [reply blog/docs/articles/knative-meets-apache-camel.md:53]: :+1:

File: blog/docs/articles/knative-meets-apache-camel.md (hunk 2 above)
Comment (creydr) [reply blog/docs/articles/knative-meets-apache-camel.md:14]: @matzew you want to address this?

**Pull request merged**